internal/provider/datasources/workspace_test.go

```diff
@@ -8,46 +8,49 @@ import (
 	"github.com/prefecthq/terraform-provider-prefect/internal/testutils"
 )
 
-func fixtureAccWorkspaceByHandle(handle string) string {
+func fixtureAccWorkspaceByHandle(workspace, handle string) string {
 	return fmt.Sprintf(`
-data "prefect_workspace" "evergreen" {
+%s
+
+data "prefect_workspace" "testdata" {
 	handle = "%s"
+	depends_on = [prefect_workspace.test]
```

Comment:
This addresses the race condition where the data source wouldn't be able to find the newly-created ephemeral workspace.